### ISSUE TITLE
Sniff array merge via plus operator when indexes are implied

### ISSUE THREAD

**Issue by rrehbein** [squizlabs/PHP_CodeSniffer#1346]
Feature request: a sniff to warn about code like the following, as it is probably a bug in code.

```php
['a list'] + ['additional list'];
```
or
```php
$aList + ['additional list'];
```

The examples should probably have been:
```php
array_merge($aList, ['additional list']);
```

**Comment by aik099** [comment]
The fixed code isn't equivalent to one, that was fixed. The PHP_CodeSniffer only does cosmetic fixes to make code look good, not to alter program logic.

As per docs the array concatenation operator (`+`) would not overwrite elements for matched keys, but `array_merge` would combine both array so that elements are preserved, but merged array indexes are lost.

For that reason I'm 👎  for this.

**Comment by rrehbeindoi** [comment]
I understand that `array_merge` and `+` behave differently.  I thought it might be something that fits in the same collection of sniffs as the `==` vs `===`.  The argument could be made that `==` and `===` are clearly documented to behave differently, therefore there should not be a sniff for it.

**Comment by jrfnl** [comment]
As long as there's no auto-fixer included in the sniff, I see nothing wrong with adding a sniff for this to `Generic`.

**Comment by gsherwood** [comment]
Closing as I'm not going to get around to adding this sniff and no additional interest has been shown.
